### PR TITLE
Cleanup bevy_winit

### PR DESCRIPTION
**Pull request by nicopap**
## Objective

The `bevy_winit` crate has a very large `lib.rs`, the size is daunting and discourages both contributors and reviewers. Beside, more code implies more ways to accidentally introduce bugs.

## Solution

This PR does a few things to reduce the line count, without changing logic:

- Move the winit event handler to a standalone function. This reduces the amount of indentation, and isolates relevant code.
- Replace the `WindowAndInputEventWriters` struct with direct calls to `World::send_event`. This is done through a new private extension trait called `AppSendEvent`. Now, instead of using `event_writers.specific_events.send(SpecificEvent { … })`, it directly does `app.send_event(SpecificEvent { … })`. Looking at the `send_event` implementation, I didn't see anything that would lead me to believe this is more costly or leads to different behaviors. With this change, we both reduce boilerplate on event sending and delete the `WindowAndInputEventWriters` struct
- Rename `window_entity` to `window`. This allows constructing most `bevy_window` events in a single line, rather than being split on several lines by `rustfmt`. This is also more consistent, since the `Entity` field of `bevy_window` events is called `window`, it makes sense to re-use the same name to designate the same thing. This removes a lot of boilerplate.
- Explicitly name the `CreateWindowParams` as a type alias instead of copy/pasting it about everywhere. In `create_windows`, instead of accepting each param field, accept the whole param. This removes a lot of boilerplate as well.

The combination of all those changes leads to a reduction of 200 lines.

### Notes to reviewers

You should really use the "Hide whitespaces" diff display mode.

The trickiest bit is probably the scale factor handling. Because we now directly access the world, I had to move event-sending code around, to avoid breaking mutual exclusion rules. I'm fairly confident it's the same behavior.

Another thing that deserves looking-at is non-linux plateform handling. I've only compiled this for linux targets, hence it might fail to compile on other plateforms.

---

## Changelog

- `bevy::window::WindowMoved`'s `entity` field has been renamed to `window`

## Migration Guide

`bevy::window::WindowMoved`'s `entity` field has been renamed to `window`. This is to be more consistent with other windowing events.

Consider changing usage:
```diff
-window_moved.entity
+window_moved.window
```

### CODE REVIEW

**Comment by alice-i-cecile** [comment]
This will conflict with #11245: opinions on merge order?

**Comment by nicopap** [comment]
This set of changes is relatively trivial. Unlike #11227, which was a pretty fundamental change in the way the event loop gets updated, this just changes the way we send event, and extracts common code in a type alias.

I'm fine if we merge #11245 first. At this point, I'm comfortable with rebasing. What I really do not want to happen is for this PR to stall like #9768. I'd be happy if this gets reviewed as soon as #11245 is merged and this rebased.

File: crates/bevy_winit/src/lib.rs
Comment (doonv) [inline crates/bevy_winit/src/lib.rs:112]: I don't really understand why you are separating the `msg` into a separate variable. If you wanted to reduce indentation I think this will work better:
```suggestion
            let app = ANDROID_APP
                .get()
                .expect("Bevy must be setup with the #[bevy_main] macro on Android")
                .clone();
            event_loop_builder.with_android_app(app);
```

**Review by doonv — APPROVED** [review]
Looks good, but it looks like `WindowTitleCache` is never used. So maybe remove that too?

File: crates/bevy_winit/src/lib.rs
Comment (nicopap) [reply crates/bevy_winit/src/lib.rs:112]: It's just a personal style choice. I prefer to avoid expanding the method chain into several lines, it occupies vertical space and you can see less things on screen at the same time.

But it's purely personal preference. If the community prefers the expanded version, I'll switch back.

Comment (doonv) [reply crates/bevy_winit/src/lib.rs:112]: Alright that's fine by me.

**Comment by alice-i-cecile** [comment]
Merging #11245 now; rebase and ping me then I'll merge this in immediately.

**Comment by nicopap** [comment]
@alice-i-cecile should be good to go.

Also TIL that you can use `-Xignore-all-space` to rebase while using a whitespace-ignoring diff for merge resolution. Without it, it would have been much harder to accomplish.

File: crates/bevy_winit/src/lib.rs
Comment (Vrixyz) [inline crates/bevy_winit/src/lib.rs:164]: I like it, and I wouldn't block on it, but this might be better as a usability method within bevy_ecs maybe?

**Review by Vrixyz — APPROVED** [review]
I like it :+1: tested a few examples on windows native + firefox

File: crates/bevy_winit/src/lib.rs
Comment (alice-i-cecile) [reply crates/bevy_winit/src/lib.rs:164]: I tried to add it earlier and it was controversial IIRC 😅 Private helper is fine though.

**Comment by alice-i-cecile** [comment]
Once we've checked that there's no new problems on Android / iOS this is good to go.

**Comment by mockersf** [comment]
blocked by #11281 and #11282 for testing

**Comment by alice-i-cecile** [comment]
Merging later today. Ping me if I forget.

**Comment by mockersf** [comment]
Oops misclicked on mobile

> Merging later today. Ping me if I forget.

Has it been tested on enough platforms?

**Comment by alice-i-cecile** [comment]
Hmm. We should resolve conflicts and do another round of testing.

**Comment by nicopap** [comment]
#11489 completes this.

**Pull request closed without merging**
